src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +498,146 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
                 return true;
             }
         }
+
+        public static bool TryGetLastDeleteDatabaseCommandWithoutErrors(ServerStore serverStore, string databaseName, out long index)
+        {
+            var commandType = nameof(DeleteDatabaseCommand);
+            index = 0;
+
+            var hasExceptions = false;
+
+            //DeleteDatabaseCommand id looks like that: DeleteDatabaseCommand/EncryptedTest_1/C/sadjghsdkjfng
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+
+                foreach (var result in table.SeekBackwardFrom(LogHistoryTable.Indexes[LogHistoryIndexSlice], Slices.AfterAllKeys))
```

Comment:
you iterate over the entire table. 
Instead let's use the known prefix here and take the max index.

Reply:
I cant because in that way they are ordered by index